ricecooker/chefs.py

```diff
@@ -0,0 +1,262 @@
+        By default this function does nothing, but subclass can use this hook to
+        run prerequisite tasks.
+            args: command line
+            options (dict): additional options passed to chef
+        """
+        pass
+
+
+    def run(self, args, options):
+        """
+        This function calls uploadchannel which performs all the run steps:
+          - Create ChannelNode
+          - Pupulate Tree with TopicNodes, ContentNodes, and associated File objects
+          - .
+          - ..
+          - ...
+
+        Args:
+            args (dict): ricecooker command line arguments
+            options (dict): additional compatibility mode options given on command line
+        """
+        self.pre_run(args, options)
+        args_and_options = args.__dict__.copy()
+        args_and_options.update(options)
+        uploadchannel(self, **args_and_options)
+
+
+    def get_channel(self, **kwargs):
+        """
+        Call chef script's get_channel method in compatibility mode
+        ...or...
+        Create a `ChannelNode` from the Chef's `channel_info` class attribute.
+
+        Args:
+            kwargs (dict): additional keyword arguments that `uploadchannel` received
+        Returns: channel created from get_channel method or None
+        """
+        if self.compatibility_mode:
+            # For pre-sushibar scritps that do not implement `get_channel`,
+            # we must check it this function exists before calling it...
+            if hasattr(self.chef_module, 'get_channel'):
+                config.LOGGER.info("Calling get_channel... ")
+                # Create channel (using the function in the chef script)
+                channel = self.chef_module.get_channel(**kwargs)
+            # For chefs with a `create_channel` method instead of `get_channel`
+            if hasattr(self.chef_module, 'create_channel'):
+                config.LOGGER.info("Calling create_channel... ")
+                # Create channel (using the function in the chef script)
+                channel = self.chef_module.create_channel(**kwargs)
+            else:
+                channel = None  # since no channel info, SushiBar functionality will be disabled...
+            return channel
+
+        elif hasattr(self, 'channel_info'):
+            # If a sublass has an `channel_info` attribute (a dict) it doesn't need
+            # to define a `get_channel` method and instead rely on this code:
+            channel = ChannelNode(
+                source_domain=self.channel_info['SOURCE_DOMAIN'],
+                source_id=self.channel_info['SOURCE_ID'],
+                title=self.channel_info['CHANNEL_TITLE'],
+                thumbnail=self.channel_info.get('THUMBNAIL'),
+            )
+            return channel
+
+        else:
+            raise NotImplementedError('BaseChef must overrride the get_channel method')
+
+
+    def construct_channel(self, **kwargs):
```

Comment:
Good idea. I guess we can make it a keyword argument `channel=None` and if you don't pass in a channel we call  `get_channel` to get it. Another option is for the first call of  `get_channel` to also attach it's result to `self.channel`. Still doesn't seem any of these options are really necessary for now so will leave this for later... I'm sure ricecooker api will evolve and this is not an issue right now.

I'm thinking users can rely on the info in `self.channel_info` (dict). If they need extensibility, they can define a `get_channel_info` method that also returns a dict (this way they don't need to know what a `ChannelNode` is to provide the info).

The sushi bar integration code can get the `channel_id` using
`uuid.uuid5(uuid.uuid5(uuid.NAMESPACE_DNS, channel_info['CHANNEL_SOURCE_DOMAIN']), channel_info['CHANNEL_SOURCE_ID']).hex`